torch/autograd/_functions/compare.py

```diff
@@ -18,7 +21,12 @@ def forward(cls, ctx, a, b):
     @staticmethod
     def backward(ctx, grad_output):
         grad_input = (grad_output * 0).type(ctx.input_type)
-        return grad_input, (grad_input if ctx.b_tensor else None)
+
+        def maybe_unexpand_or_view_if_tensor(tensor, size):
+            return tensor if (tensor is None or size is None) else maybe_unexpand_or_view(tensor, size)
```

Comment:
do you really need to define a function for it? it's only used once, so you can just break it down into multiple steps that compute the full expression